src/Core/Collision/PolygonalCollider.cpp

```diff
@@ -104,25 +104,21 @@ namespace obe::Collision
             }
         }
 
-        const Transform::UnitVector destPos
-            = (this->getCentroid() + offset).to<Transform::Units::ScenePixels>();
         if (!reachableColliders.empty())
         {
-            std::pair<double, Transform::UnitVector> minDist(-1, Transform::UnitVector());
-            for (auto& reachable : reachableColliders) {
-                double dist = std::sqrt(std::pow(reachable.second.x - destPos.x, 2)
-                    + std::pow(reachable.second.y - destPos.y, 2));
-                if (minDist.first == -1 || minDist.first > dist)
+            // Get lowest offset between this collider and a reachable collider
+            for (auto& reachable : reachableColliders)
+            {
+                if (reachable.second.magnitude() < collData.offset.to(Transform::Units::ScenePixels).magnitude())
```

Comment:
Why do you need to convert to ScenePixels here ?

Reply:
Because `getMaximumDistanceBeforeCollision(PolygonalCollider& collider, const Transform::UnitVector& offset)` return a vector in ScenePixels unit.

Reply:
Ah okay !